### ISSUE TITLE
[Server] Send all provisioning events

### ISSUE THREAD

**Issue by leecalcote** [meshery/meshery#8771]
> @zakisk we're not getting all of Meshery Server's errors sent to the /apis/events, are we?
> 
> For instance, I see this error on stdout, but not via Meshery UI:
> 
> ```
> ERRO[2023-09-13T16:07:12-05:00] the platform local is not supported currently. The supported platforms are: Docker and Kubernetes  app=meshery code=1231 probable-cause="Your platform is not supported for deploying Meshery Adapters" severity=3 short-description="Unable to deploy Meshery Adapter in the current environment" suggested-remediation="Consider using a supported platform for deploying Meshery Adapters"
> ```
> 
> I wonder if you can help with this in a separate PR.
> 
> @MUzairS15 guidance?

@zakisk Refer other parts of server for eg: in design_engine_handler.go or k8s_config_handler.go for sending events

_Originally posted by @MUzairS15 in https://github.com/meshery/meshery/issues/8694#issuecomment-1719607280_

**Comment by Jougan-0** [comment]
@leecalcote can i have a go on this issue

**Comment by zakisk** [comment]
@MUzairS15 @leecalcote  at [this line](https://github.com/meshery/meshery/blob/6b66146f32fd7d39bd65ac8d449d99d52d8aeaca/server/handlers/k8sconfig_handler.go#L94) category and action are set to connection and create respectively, if I were to send an event for wrong platform error what category and action do I set? would platform and deploy be good? or do we have predefined categories and actions?

**Comment by EraKin575** [comment]
I would like to work on this

**Comment by Jougan-0** [comment]
@leecalcote can I work on this ??

**Comment by leecalcote** [comment]
@zakisk, here are the categories and actions that have been used thus far. You should find that these event types fall into an existing combination, if you don't, please suggest a new one.

**event.category:**
"user"
"signup_request"
"badge"
"github_artifacts"
"github_action"
"catalog_request"
"application"
"token"
"credential"
"user_role"
"smi_result"
"filter"
"user_preference"
"connection"
"performance_result"
"design"
"github_artifact"
"pattern"
"performance_profile"

**event.action:**
"assign"
"login"
"remove_from_organization"
"import"
"deny"
"register"
"clone"
"approve"
"upload"
"unpublish"
"configure"
"delete"
"update"
"activity"
"create"
"unpubish_request"
"signup"
"add_to_organization"
"invite"
"return"
"share"
"publish"

**Comment by leecalcote** [comment]
@zakisk will you add this list to a Contributing Doc on Events?

**Comment by zakisk** [comment]
@leecalcote yes

**Comment by zakisk** [comment]
@leecalcote do I need to send all provisioning events in all handlers in this [directory](https://github.com/meshery/meshery/tree/master/server/handlers).

**Comment by leecalcote** [comment]
@zakisk, yes, please. Might be an exception or two, however, the overwhelming majority of the time we're going to want to register an event.

**Comment by vansh106** [comment]
Is this issue still open ? 
Can I work on this

**Comment by saurabh100ni** [comment]
@vansh106 you still up for this issue?

**Comment by vansh106** [comment]
Yes I am, I was busy for a while, I ll start working now.Sent from my iPhoneOn 08-Feb-2024, at 7:18 PM, Saurabh Soni ***@***.***> wrote:﻿
@vansh106 you still up for this issue?

**Comment by saurabh100ni** [comment]
Good deal @vansh106 , in this case I am assigning you again.

**Comment by Tharanishwaran** [comment]
@saurabh100ni i like to work on this issue can you assign me to this issue ?

**Comment by tuningfolk** [comment]
I would like to work on this issue. Could this be assigned to me?

**Comment by Bhumikagarggg** [comment]
@tuningfolk any updates?
